### PR TITLE
refactor(experimental): add text-encoding-impl package

### PR DESCRIPTION
**Pull request by lorisleiva**
Abstract the implementation of TextEncoder and TextDecoder which are available on Node and Browser environments but not on React Native. For the latter the most performant polyfill library `fastestsmallesttextencoderdecoder` is used as a peer dependency.

### CODE REVIEW

**Comment by lorisleiva** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1630** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1630" target="_blank">Graphite</a> 
    * **PR #1631** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1631" target="_blank">Graphite</a> 
      * **PR #1632** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1632" target="_blank">Graphite</a> 
        * **PR #1633** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1633" target="_blank">Graphite</a> 
          * **PR #1639** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1639" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1639?utm_source=stack-comment).

File: packages/text-encoding-impl/package.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        "eslint-plugin-sort-keys-fix": "^1.1.2",
+        "jest": "^29.7.0",
+        "jest-environment-jsdom": "^29.6.4",
+        "jest-runner-eslint": "^2.1.0",
+        "jest-runner-prettier": "^1.0.0",
+        "prettier": "^2.8",
+        "test-config": "workspace:*",
+        "tsconfig": "workspace:*",
+        "tsup": "7.2.0",
+        "typescript": "^5.1.6"
```
Comment (steveluscher) [inline packages/text-encoding-impl/package.json:64]:
```suggestion
        "typescript": "^5.2.2"
```

File: packages/text-encoding-impl/package.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    "peerDependencies": {
+        "fastestsmallesttextencoderdecoder": "^1.0.22"
```
Comment (steveluscher) [inline packages/text-encoding-impl/package.json:67]: (Y)

**Review by steveluscher — APPROVED** [review]
[yesyes](https://app.graphite.dev/macros/solana-labs)

File: packages/text-encoding-impl/package.json (hunk 1 above)
Comment (lorisleiva) [reply packages/text-encoding-impl/package.json]: Done ✅

**Comment by lorisleiva** [comment]
### Merge Activity

* **Sep 30, 6:16 PM**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1639).

**Pull request merged**